test/robot_test.coffee

```diff
@@ -35,6 +35,7 @@ describe 'Robot', ->
 
     # Re-throw AssertionErrors for clearer test failures
     @robot.on 'error', (name, err, response) ->
+      return unless err?
```

Comment:
Hmmm.. I'm very confused by this. First, we have `robot.error != robot.on 'error'`, and then the funkiness around arity. I honestly don't remember what was going on, but I thought `name` was needed for some reason... Error handling probably needs a review pass and cleanup